src/main/java/nextstep/section/entity/Section.java

```diff
@@ -0,0 +1,125 @@
+package nextstep.section.entity;
+
+import nextstep.section.exception.SectionException;
+import nextstep.station.entity.Station;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Objects;
+
+import static nextstep.common.constant.ErrorCode.SECTION_DISTANCE_TOO_SHORT;
+
+@Entity
+public class Section {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "UP_STATION_ID")
+    private Station upStation;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "DOWN_STATION_ID")
+    private Station downStation;
+
+    @NotNull
+    private Long distance;
+
+    @OneToOne(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    @JoinColumn(name = "PREVIOUS_SECTION_ID")
+    private Section previousSection;
+
+    @OneToOne(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    @JoinColumn(name = "NEXT_SECTION_ID")
+    private Section nextSection;
+
+    public Section() {
+    }
+
+    public Section(Long id, Station upStation, Station downStation, Long distance) {
+        this.id = id;
+        this.upStation = upStation;
+        this.downStation = downStation;
+        this.distance = distance;
+    }
+
+    public static Section of(final Long id, final Station upStation, final Station downStation, final Long distance) {
+        if (distance < 1) {
+            throw new SectionException(String.valueOf(SECTION_DISTANCE_TOO_SHORT));
+        }
+
+        return new Section(id, upStation, downStation, distance);
+    }
+
+    public static Section of(final Station upStation, final Station downStation, final Long distance) {
+        if (distance < 1) {
+            throw new SectionException(String.valueOf(SECTION_DISTANCE_TOO_SHORT));
+        }
+
+        return new Section(null, upStation, downStation, distance);
```

Comment:
기존에 정의된 정적 팩토리 메서드를 호출한다면, 검증 로직 중복을 줄여볼 수 있지 않을까요?
```suggestion
        return of(null, upStation, downStation, distance);
```